src/deser.rs

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::io::Read;
+
+use serde::Serialize;
+use serde::de::DeserializeOwned;
+
+use ron::ser::pretty::to_string as to_ron_string;
+use ron::de::from_reader as from_ron_string;
+
+#[cfg(feature = "yaml")]
+use serde_yaml::{to_string as to_yaml_string, from_reader as from_yaml_string};
+
+/// A trait to bundle serializer and deserializer
+pub trait DeSerializer<T: Serialize + DeserializeOwned> {
```

Comment:
TBH I don't understand why we do not have such a thing in serde itself.

Reply:
I'm not sure :P I guess 'bundle' traits are just code duplication at the end